### PR TITLE
feat(php-buildpack): add AMQP extension

### PR DESCRIPTION
**Pull request by Frzk**
Add support for AMQP extension.

### CODE REVIEW

File: support/ext/amqp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+set -ex
+
+
+download_archive() {
+    local archive_url
+
+    archive_url="${1}"
+
+    curl -L "${archive_url}"
```
Comment (EtienneM) [inline support/ext/amqp:11]: I prefer we use the long option in shell script. It's often more understand-able
```suggestion
    curl --location "${archive_url}"
```

**Review by EtienneM — CHANGES_REQUESTED** [review]

File: support/ext/amqp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+# set it to /app/vendor/php/lib/rabbitmq.
+# This means `amqp.so` will look for the `librabbitmq.so` file in
+# /app/vendor/php/lib/rabbitmq/lib/
+#
+# We specify the same path when compiling amqp (`--with-librabbitmq-dir`).
+#
+# Since librabbitmq must be deployed along with the amqp extension, it must be
+# added to the future package. That's why we are moving the librabbitmq files
+# to $PREFIX once they are ready.
+#
```
Comment (EtienneM) [inline support/ext/amqp:20]: Thanks for this comprehensive comment 👍🏾

**Review by EtienneM — COMMENTED** [review]
You should rebase with master, you apparently have conflict in the conf/buildpack.conf file. I usually execute `git pull --rebase origin master` to solve it.

Otherwise it looks good to me, hoping it works 🤞🏾

**Review by EtienneM — APPROVED** [review]

**Comment by Frzk** [comment]
Thanks for the tip.
I'll wait feedback from the client before merging.

**Comment by johnsudaar** [comment]
Extension built for PHP 7.4 too

**Comment by johnsudaar** [comment]
The script break if you run it twice (with 2 different PHP API version):

```
mv: cannot move '/app/vendor/php/lib/rabbitmq/lib/x86_64-linux-gnu/pkgconfig' to '/app/vendor/php/lib/rabbitmq/lib/pkgconfig': Directory not empty                                                                                                                                                                                                         
mv: cannot move '/app/vendor/php/lib/rabbitmq/lib/x86_64-linux-gnu/cmake' to '/app/vendor/php/lib/rabbitmq/lib/cmake': Directory not empty   
```

File: support/ext/amqp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+pkgname="amqp"
+version="${amqp_version}"
+source_url="https://pecl.php.net/get/amqp-%s.tgz"
+
+build_amqp "${pkgname}" "${version}" "${source_url}" "${librabbitmq_prefix}"
+
+cp "${pkgname}-${version}/modules/amqp.so" "${EXT_DIR}/amqp.so"
+echo "extension=amqp.so" > "${PREFIX}/etc/conf.d/amqp.ini"
+
```
Comment (johnsudaar) [inline support/ext/amqp:157]: Could you write a cleanup script that remove all the temporary directories and file used ?

**Review by johnsudaar — CHANGES_REQUESTED** [review]
One small  change but otherwise it looks really promising.

**Comment by johnsudaar** [comment]
The client tested it and it seems to work fine! GG !

**Comment by btrd** [comment]
Link to https://github.com/Scalingo/php-buildpack/issues/258

File: support/ext/amqp (hunk 3 above)
Comment (Frzk) [reply support/ext/amqp:178]: Should be fixed by https://github.com/Scalingo/php-buildpack/pull/262/commits/b3bd78b5208a90345b50aa90aa6480a328e92955.

**Review by johnsudaar — APPROVED** [review]
LGTM

**Review by EtienneM — APPROVED** [review]

**Pull request merged**